crates/commitlog/src/segment.rs

```diff
@@ -22,23 +22,28 @@ pub const MAGIC: [u8; 6] = *b"(ds)^2";
 pub struct Header {
     pub log_format_version: LogFormatVersion,
     pub checksum_algorithm: ChecksumAlgorithm,
+    pub compression: SegmentCompression,
 }
 
 #[derive(Clone, Copy, Debug, Eq, PartialEq, Ord, PartialOrd)]
 #[repr(u8)]
 pub enum LogFormatVersion {
     V0,
     V1,
+    V2,
```

Comment:
I don't think we need a new version -- zstd has a magic number which we can expect to appear before the commitlog magic number in a stream.

Reply:
Yes, take a look at https://github.com/clockworklabs/SpacetimeDB/pull/2034/files#diff-e4988280b2a00468abcbf980d6024d0f07d179b322c732057bb6d74f5f3583aaR31